### ISSUE TITLE
Error occurs when building from source:  /usr/bin/ld: cannot open output file lightseq/inference.cpython-37m-x86_64-linux-gnu.so: No such file or directory collect2: error: ld returned 1 exit status

### ISSUE THREAD

**Issue by wu1143100799** [bytedance/lightseq#140]
[ 74%] Linking CUDA device code CMakeFiles/lightseq.dir/cmake_device_link.o
[ 75%] Linking CXX shared module lightseq/inference.cpython-37m-x86_64-linux-gnu.so
/usr/bin/ld: cannot open output file lightseq/inference.cpython-37m-x86_64-linux-gnu.so: No such file or directory
collect2: error: ld returned 1 exit status
lightseq/inference/pywrapper/CMakeFiles/lightseq.dir/build.make:142: recipe for target 'lightseq/inference/pywrapper/lightseq/inference.cpython-37m-x86_64-linux-gnu.so' failed
make[2]: *** [lightseq/inference/pywrapper/lightseq/inference.cpython-37m-x86_64-linux-gnu.so] Error 1
CMakeFiles/Makefile2:553: recipe for target 'lightseq/inference/pywrapper/CMakeFiles/lightseq.dir/all' failed
make[1]: *** [lightseq/inference/pywrapper/CMakeFiles/lightseq.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 77%] Linking CUDA device code CMakeFiles/transformer_example.dir/cmake_device_link.o
[ 80%] Linking CUDA device code CMakeFiles/transformer_decoder_example.dir/cmake_device_link.o
[ 80%] Linking CUDA device code CMakeFiles/transformer_generate_example.dir/cmake_device_link.o
[ 82%] Linking CXX executable transformer_example
[ 85%] Linking CXX executable transformer_generate_example
[ 87%] Linking CUDA device code CMakeFiles/transformer_generate_server.dir/cmake_device_link.o
[ 87%] Linking CXX executable transformer_decoder_example
[ 90%] Linking CUDA device code CMakeFiles/decoder_generate_server.dir/cmake_device_link.o
[ 90%] Linking CUDA device code CMakeFiles/transformer_server.dir/cmake_device_link.o
[ 90%] Built target transformer_example
[ 90%] Built target transformer_generate_example
[ 90%] Built target transformer_decoder_example
[ 91%] Linking CXX shared library libtransformer_generate_server.so
[ 95%] Linking CXX shared library libdecoder_generate_server.so
[ 95%] Linking CXX shared library libtransformer_server.so
[100%] Built target transformer_generate_server
[100%] Built target transformer_server
[100%] Built target decoder_generate_server
Makefile:90: recipe for target 'all' failed
make: *** [all] Error 2

<img width="1330" alt="截屏2021-07-30 上午9 03 46" src="https://user-images.githubusercontent.com/48854292/127584599-e1f80e77-33c4-4a87-bc68-bcccb76295e4.png">
<img width="1326" alt="截屏2021-07-30 上午9 04 21" src="https://user-images.githubusercontent.com/48854292/127584626-f6968b63-989a-41a3-b649-ec8f31585b74.png">
<img width="1200" alt="截屏2021-07-30 上午9 04 56" src="https://user-images.githubusercontent.com/48854292/127584650-9fc08be0-85eb-4da6-bb55-83d04fab37cd.png">
<img width="1333" alt="截屏2021-07-30 上午9 05 21" src="https://user-images.githubusercontent.com/48854292/127584673-21be6000-1b73-40cb-b5af-4b99d97b5bae.png">

Environments:
  cuda-10.2
  protobuf-3.13.0
  hdf5-1.12.1
  cmake-3.21.1

I wonder how to solve it.

**Comment by Taka152** [comment]
I guess their is already an `inference.cpython.*.so` file in the `lightseq/` directory.

If true, you can delete it and try.

**Comment by wu1143100799** [comment]
However, I don't find an **inference.cpython.*.so file** in the lightseq/ directory. 
I download the source code again and meet the same error when building.

**Comment by Taka152** [comment]
Did you download the code on master branch?
________________________________
发件人: wu1143100799 ***@***.***>
发送时间: Friday, July 30, 2021 5:48:39 PM
收件人: bytedance/lightseq ***@***.***>
抄送: Ying Xiong ***@***.***>; Comment ***@***.***>
主题: Re: [bytedance/lightseq] Error occurs when building from source: /usr/bin/ld: cannot open output file lightseq/inference.cpython-37m-x86_64-linux-gnu.so: No such file or directory collect2: error: ld returned 1 exit status (#140)


However, I don't find an inference.cpython.*.so file in the lightseq/ directory.
I download the source code again and meet the same error when building.

―

**Comment by wu1143100799** [comment]
I  download the code from both master branch and  release 2.1.0 source-code.zip and  both meet the same error.  
Does the  **inference.cpython.*.so** file  generate when building?

**Comment by Taka152** [comment]
Yes, it should be generate. Building works fine in my env on master branch.
________________________________
发件人: wu1143100799 ***@***.***>
发送时间: Saturday, July 31, 2021 9:55:54 AM
收件人: bytedance/lightseq ***@***.***>
抄送: Ying Xiong ***@***.***>; Comment ***@***.***>
主题: Re: [bytedance/lightseq] Error occurs when building from source: /usr/bin/ld: cannot open output file lightseq/inference.cpython-37m-x86_64-linux-gnu.so: No such file or directory collect2: error: ld returned 1 exit status (#140)


I download the code from both master branch and release 2.1.0 source-code.zip and both meet the same error.
Does the inference.cpython.*.so file generate when building?

―

**Comment by Taka152** [comment]
You can also try building from the source in this docker image
https://github.com/bytedance/lightseq/blob/master/docker/Dockerfile.pypi
